### PR TITLE
external-service-doc

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1345

### CODE REVIEW

**Review by willowmck — COMMENTED** [review]
Just one small typo, but looks good otherwise!

File: docs/content/guides/external_services.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+## Declaring External Services
+
+#### Motivation
+
+Sometimes within a mesh, an application needs the ability to communicate with services which are "external" to that mesh. For instance a public API, or other external API providers. Istio exposes this functionality via their [`ServiceEntry`](https://istio.io/latest/docs/reference/config/networking/service-entry/) API. Gloo Mesh exposes the same functionality, in a way that feels natural within the rest of the Gloo Mesh API ecosystem.
+
+#### Destination CRD
+
+The [Destination CRD]({{% versioned_link_path fromRoot="/reference/api/github.com.solo-io.gloo-mesh.api.networking.v1.destination/" %}}) can be thought of as a host. In other words an addressable host which may be backed by 0-n endpoints. Typically these `Destinations` are Discovered by Gloo Mesh, and are backed by kubernetes workloads. However, that is only 1 use case. An external service at it's core is just a hostname backed my 0-n endpoints as well, so in comes the `External Destination`.
```
Comment (harveyxia) [inline docs/content/guides/external_services.md:28]:
```suggestion
The [Destination CRD]({{% versioned_link_path fromRoot="/reference/api/github.com.solo-io.gloo-mesh.api.discovery.v1.destination/" %}}) can be thought of as a host. In other words an addressable host which may be backed by 0-n endpoints. Typically these `Destinations` are Discovered by Gloo Mesh, and are backed by kubernetes workloads. However, that is only 1 use case. An external service at it's core is just a hostname backed my 0-n endpoints as well, so in comes the `External Destination`.
```

**Review by harveyxia — APPROVED** [review]

File: docs/content/guides/external_services.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+The [Destination CRD]({{% versioned_link_path fromRoot="/reference/api/github.com.solo-io.gloo-mesh.api.discovery.v1.destination/" %}}) can be thought of as a host. In other words an addressable host which may be backed by any number of endpoints. Typically these destinations are discovered by Gloo Mesh and are backed by Kubernetes workloads. However, an external service represents a destination external to the service mesh, and perhaps the Kubernetes cluster. The Destination CRD provides a way to represent these external services manually.
+
+The `External Destination` is not a seperate CRD in itself, but rather a way to configure the `Destination` CRD to accomplish our goal of routing to services which exist outside of our mesh.
+
+### Demo
+
+Enough explaining, let's get to a quick example.
+
+To start let's create our external service. Simply apply the following yaml to the cluster:
```
Comment (rvennam987) [inline docs/content/guides/external_services.md:36]: to the Gloo Mesh management cluster

Comment (rvennam) [inline docs/content/guides/external_services.md:36]: to the Gloo Mesh management cluster:

**Pull request merged**